backend/inspirehep/records/api/literature.py

```diff
@@ -484,6 +487,14 @@ def add_file(
             filename = filename or key
             if not filename:
                 filename = new_key
+            if mimetype in RESTRICTED_MIMETYPES:
+                LOGGER.error(
+                    "Unsupported file type - Aborting",
+                    key=key,
+                    mimetype=mimetype,
+                    thread=threading.get_ident(),
+                )
+                raise UnsupportedFileError(mimetype)
```

Comment:
This will cause a 500 and display a generic message in the editor. I think it would be good to expose this to the API (the invenio way is to subclass `invenio-rest.RESTException`, not sure that's how it should be done here). @drjova wdyt?

Reply:
I updated `UnsupportedFileError` definition. It is a subclass of `RESTException` now. I think it should work.

Reply:
Inheriting `RESTException` should be enough